nebula-logger/tests/logger-engine/classes/LogEntryEventBuilder_Tests.cls

```diff
@@ -374,10 +383,246 @@ private class LogEntryEventBuilder_Tests {
+
+        Test.startTest();
+        builder.setDatabaseResult(undeleteResults);
+        Test.stopTest();
+
+        LogEntryEvent__e logEntryEvent = builder.getLogEntryEvent();
+
+        System.assertEquals('List', logEntryEvent.DatabaseResultCollectionType__c);
+        System.assertEquals(JSON.serializePretty(undeleteResults), logEntryEvent.DatabaseResultJson__c);
+        System.assertEquals(Database.UndeleteResult.class.getName(), logEntryEvent.DatabaseResultType__c);
+    }
+
+    @isTest
+    static void it_should_set_database_result_fields_for_list_of_failed_undeleteResult() {
+        List<Log__c> logs = new List<Log__c>();
+        for (Integer i = 0; i < 5; i++) {
+            Log__c log = new Log__c(TransactionId__c = '1234' + i);
+            logs.add(log);
+        }
+        insert logs;
+        delete logs;
+        // Hard delete to (later) cause an undelete error
+        Database.emptyRecycleBin(logs);
+
+        List<Database.UndeleteResult> undeleteResults = Database.undelete(logs, false);
+        for (Database.UndeleteResult undeleteResult : undeleteResults) {
+            System.assertEquals(false, undeleteResult.isSuccess());
+        }
+
+        LogEntryEventBuilder builder = new LogEntryEventBuilder(LoggingLevel.INFO, true);
+
+        Test.startTest();
+        builder.setDatabaseResult(undeleteResults);
+        Test.stopTest();
+
+        LogEntryEvent__e logEntryEvent = builder.getLogEntryEvent();
+
+        System.assertEquals('List', logEntryEvent.DatabaseResultCollectionType__c);
+        System.assertEquals(JSON.serializePretty(undeleteResults), logEntryEvent.DatabaseResultJson__c);
+        System.assertEquals(Database.UndeleteResult.class.getName(), logEntryEvent.DatabaseResultType__c);
+    }
+
+    @isTest
+    static void it_should_set_database_result_fields_for_list_of_successful_upsertResult_when_insert() {
+        // UpsertResult for a new record insert
+        List<Log__c> logs = new List<Log__c>();
+        for (Integer i = 0; i < 5; i++) {
+            Log__c log = new Log__c(TransactionId__c = '1234' + i);
+            logs.add(log);
+        }
+        List<Database.UpsertResult> upsertResults = Database.upsert(logs, false);
+        for (Database.UpsertResult upsertResult : upsertResults) {
+            System.assertEquals(true, upsertResult.isSuccess());
+        }
+
+        LogEntryEventBuilder builder = new LogEntryEventBuilder(LoggingLevel.INFO, true);
+
+        Test.startTest();
+        builder.setDatabaseResult(upsertResults);
+        Test.stopTest();
+
+        LogEntryEvent__e logEntryEvent = builder.getLogEntryEvent();
+
+        System.assertEquals('List', logEntryEvent.DatabaseResultCollectionType__c);
+        System.assertEquals(JSON.serializePretty(upsertResults), logEntryEvent.DatabaseResultJson__c);
+        System.assertEquals(Database.UpsertResult.class.getName(), logEntryEvent.DatabaseResultType__c);
```

Comment:
mmm, this is an interesting one. I guess that we just say, hey, the JSON is there in the event that you need to examine the results of `isCreated()` for each result